vector/src/main/java/im/vector/riotx/features/home/UnknwonDeviceDetectorSharedViewModel.kt

```diff
@@ -76,12 +97,28 @@ class UnknownDeviceDetectorSharedViewModel(session: Session, initialState: Unkno
                 }
     }
 
-    override fun handle(action: EmptyAction) {}
+    override fun handle(action: Action) {
+        when (action) {
+            is Action.IgnoreDevice -> {
+                // local echo
+                withState { state ->
+                    state.unknownSessions.invoke()?.let {
+                        val updated = it.filter { it.second.deviceId != action.deviceId  }
+                        setState {
+                            copy(unknownSessions = Success(updated))
+                        }
+                    }
+                }
+                ignoredDeviceList.add(action.deviceId)
+                vectorPreferences.storeUnknownDeviceDismissedList(ignoredDeviceList)
+            }
+        }
+    }
 
     companion object : MvRxViewModelFactory<UnknownDeviceDetectorSharedViewModel, UnknownDevicesState> {
         override fun create(viewModelContext: ViewModelContext, state: UnknownDevicesState): UnknownDeviceDetectorSharedViewModel? {
             val session = (viewModelContext.activity as HasScreenInjector).injector().activeSessionHolder().getActiveSession()
-            return UnknownDeviceDetectorSharedViewModel(session, state)
+            return UnknownDeviceDetectorSharedViewModel(session, VectorPreferences(viewModelContext.activity()), state)
```

Comment:
I think this is not the correct way to pass the VectorPreferences here. It should be injected in the ViewModel's constructor. See for instance RoomDetailViewModel

Reply:
I didn't wanted to add the full framework, just to create a view model
Will see in next PR, as it will be reworked (no more one popup per device)